mitosheet/mitosheet/saved_analyses/upgrade.py

```diff
@@ -317,5 +318,8 @@ def upgrade_saved_analysis_to_current_version(saved_analysis: Optional[Dict[str,
     new_format_saved_analysis = upgrade_saved_analysis_format_to_steps_data(saved_analysis)
     saved_analysis = upgrade_steps_for_new_format(new_format_saved_analysis)
 
+    if saved_analysis is not None and 'public_interface_version' not in saved_analysis:
+        saved_analysis['public_interface_version'] = 1
```

Comment:
TODO: we have to show a different version of the interface if we're on the old one.